automation_tools/__init__.py

```diff
@@ -1467,3 +1471,246 @@ def setenforce(mode):
+    image = openstack_client.images.find(name=image_name)
+    flavor = openstack_client.flavors.find(name=flavor_name)
+    network = openstack_client.networks.find(label=network_name)
+    floating_ip = openstack_client.floating_ips.create(
+        openstack_client.floating_ip_pools.list()[0].name
+    )
+    # Create instance from the given parameters
+    instance = openstack_client.servers.create(
+        name=instance_name,
+        image=image.id,
+        flavor=flavor.id,
+        key_name=ssh_key,
+        network=network.id
+    )
+    # Assigning floating ip to instance
+    instance_status = False
+    while instance_status is not True:
+        try:
+            instance.add_floating_ip(floating_ip)
+        except novaclient.exceptions.BadRequest:
+            instance_status = False
+            time.sleep(5)
+        else:
+            instance_status = True
+    # Wait till DNS resolves the IP
+    time.sleep(600)
+    # Getting Hostname from IP
+    env['instance_host'] = get_hostname_from_ip(str(floating_ip.ip))
+
+
+def delete_openstack_instance(instance_name):
+    """Deletes openstack Instance.
+
+    :param instance_name: A string. Openstack instance name to delete.
+
+    The following environment variables affect this command:
+
+    USERNAME
+        The username of an openstack project to login.
+    PASSWORD
+        The password of an openstack project to login.
+    AUTH_URL
+        The authentication url of the project.
+    PROJECT_ID
+        Project ID of an openstack project.
+
+    """
+    openstack_client = get_openstack_client()
+    try:
+        instance = openstack_client.servers.find(name=instance_name)
+    except novaclient.exceptions.NotFound:
+        print('Instance {0} not found in Openstack project.'.format(
+            instance_name
+        ))
+        return
+    instance.delete()
+
+
+def satellite6_upgrade(admin_password=None):
+    """Upgrades satellite from already created Openstack image
+    of old Satellite version to latest Satellite version compose.
+
+    The following environment variables affect this command:
+
+    ADMIN_PASSWORD
+        Optional, defaults to 'changeme'. Foreman admin password.
+    BASE_URL
+        URL for the compose repository.
+
+    """
+    base_url = os.environ.get('BASE_URL')
+    if base_url is None:
+        print('The BASE_URL environment variable should be defined')
+        sys.exit(1)
+    if admin_password is None:
+        admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
+    # Removing rhel-released and rhel-optional repo
+    run('rm -rf /etc/yum.repos.d/rhel-{optional,released}.repo')
+    # Update the packages
+    update_packages(quiet=True)
+    # Setting Satellite61 Repos
+    repos = [
+        'rhel-{0}-server-satellite-6.1-rpms',
+        'rhel-{0}-server-satellite-6.0-rpms'
+    ]
+    enable_repos(repos[0].format(distro_info()[1]))
+    disable_repos(repos[1].format(distro_info()[1]))
```

Comment:
Awesome, and that is required when I will deal with RHEL6 network configuration.
Updated!